### PR TITLE
add TV Tower hud element

### PR DESCRIPTION
**Pull request by Colin-Blankenburg**
Screenshot 2024-10-06 182318

### CODE REVIEW

**Comment by Colin-Blankenburg** [comment]
good enough

**Comment by Colin-Blankenburg** [comment]
I got a C 😃

**Comment by Yuvix25** [comment]
@Colin-Blankenburg How about an A? 😏

**Comment by Colin-Blankenburg** [comment]
lol

File: Pages/Settings.cshtml [hunk 1]
```diff
@@ -58,6 +58,15 @@
 
     <slider-setting key="positionBarCellCount" min="3" max="51" step="2" value="13">Position Bar Cell Count</slider-setting>
 
+    <slider-setting key="tvTowerMaxSize" min="10" max="36" step="2" value="16">TV Tower Maximum Player Count</slider-setting>
+    <toggle-setting key="tvTowerCarLogo">Show TV Tower Image</toggle-setting>
```
Comment (Yuvix25) [inline Pages/Settings.cshtml:62]: Move this setting one row below so that the choice setting is right after it.

File: wwwroot/css/app.css [hunk 2]
```diff
@@ -954,7 +1108,7 @@ progress#revs::-webkit-progress-value {
 #rake-bar-pointer {
   width: 3%;
   height: 100%;
-  background-color: white;
+  background-color: rgb(0, 0, 0);
```
Comment (Yuvix25) [inline wwwroot/css/app.css:1111]: Looks better I guess?

File: wwwroot/ts/hudElements/TvTower.ts [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            const ratingElement: HTMLSpanElement = tvTowerCell.querySelector('.tv-tower-driver-rating');
+            const reputationElement: HTMLSpanElement = tvTowerCell.querySelector('.tv-tower-driver-reputation');
+            const laptimeElement: HTMLSpanElement = tvTowerCell.querySelector('.tv-tower-driver-laptime');
+            const deltaElement: HTMLSpanElement = tvTowerCell.querySelector('.tv-tower-driver-delta');
+
+            if (start <= i && i <= end) {
+
+            // `currentDriverUpdate` later can converge from `i` to allow scrolling  
+            if(i == 0) {
+            currentDriverUpdate = i;
```
Comment (Yuvix25) [inline wwwroot/ts/hudElements/TvTower.ts:143]: Indentation!

File: wwwroot/ts/hudElements/TvTower.ts [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+            }
+
+
+            //assigning 1st cell if mandatory -> out of range of P1
+            if(i == 0 && mandatoryFirstPlace) {
+                currentDriverUpdate = 0;
+            }
+
+
+                const driver = driverData[currentDriverUpdate];
```
Comment (Yuvix25) [inline wwwroot/ts/hudElements/TvTower.ts:170]: Indentation!

File: wwwroot/ts/hudElements/TvTower.ts [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+                    laptimeElement.textContent = '';
+                }
+
+                if (deltaString != null && (Driver.mainDriver.userId !== driver.driverInfo.userId.toString() || showDeltaForMainDriver)) {
+                    if (deltaNumber != null) deltaString = deltaNumber > 0 ? `+${deltaString}` : deltaString;
+
+                    if(!isDriver) { 
+                    deltaElement.textContent = deltaString;
+                    } else { 
+                    deltaElement.textContent = '';
```
Comment (Yuvix25) [inline wwwroot/ts/hudElements/TvTower.ts:312]: Indentation!

**Review by Yuvix25 — COMMENTED** [review]
Looks good overall, a few minor comments.
Much better than your last PR though :)

As for the indentation comments - in VS Code you can use Ctrl+Shift+F to auto-format a document. Make sure you only do that in new files (that you've added), because if you do it in existing files it will cause lots of new changes which are unrelated to your commit.

**Pull request merged**